api/envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -126,6 +126,9 @@ message Permission {
     // match, this permission would not match. Conversely, if the value of `not_rule` would not
     // match, this permission would match.
     Permission not_rule = 8;
+
+    // The name of the destination server that the client is attempting to connect to.
+    string requested_server_name = 9;
```

Comment:
How about `requested_sni_server_name`?

Reply:
I think we made name change from SNI to server names in FilterChainMatch, so it is not TLS specific? If here we're going to match same thing it shouldn't refer SNI directly. ref: #3454
cc @ggreenway @PiotrSikora

Reply:
That's correct. It was named this way so that it could also apply to HTTP Host header, or any other similar field in another protocol.

Reply:
Eh? RBAC already has dedicated fields for HTTP host. The fact that this specific thing applies to the SNI value presented in the ClientHello message means we need to be very explicit about what server name this refers to. If consistency is needed, then we need to stick with requested_Server_name. But as @mattklein123 points out, its still confusing, because there are one too many server names :). 
How about tls_server_name ? Very little scope for confusion. I really don't want to give people the impression that this could *also*  be a HTTP host header, as it receives separate treatment in RBAC.

Reply:
If we keep `requested_server_name`, my ask is that we make the documentation much more explicit. Basically call out that this is typically set via TLS SNI, and then explain how it might be set some other way (Listener filter, other?). Thank you.

Reply:
I don't think it could apply to HTTP Host header, but definitely similar field in another protocol. Any listener filter extension may set that. TLS inspector is the only one setting that here, but we may have other. +1 to https://github.com/envoyproxy/envoy/pull/4662/files#r224160788 make documentation explicit.

Reply:
Addressed in 92e54f4.